ingest/mongo_connection.py

```diff
@@ -42,27 +42,33 @@ def graceful_auto_reconnect(mongo_op_func):
     """Gracefully handles a reconnection event as well as other exceptions
         for mongo.
     """
+    MAX_ATTEMPTS = 5
 
-    MAX_AUTO_RECONNECT_ATTEMPTS = 5
+    def retry(attempt_num):
+        if attempt_num < MAX_ATTEMPTS - 1:
+            wait_time = 0.5 * pow(2, attempt_num)  # exponential back off
```

Comment:
Blocker: please add [jitter](https://www.baeldung.com/resilience4j-backoff-jitter#jitter) to avoid a [thundering herd](https://en.wikipedia.org/wiki/Thundering_herd_problem) of synchronous (yet exponentially-backed-off) bulk writes across jobs, e.g. in case many jobs are launched in parallel.